Jenkinsfile

```diff
@@ -81,5 +89,44 @@ pipeline {
             cleanWs()
         }
     }
+}
+
+def getSemverVersion() {
+    def nextSemver = edgeXSemver 'init'
+   //  println "--> Next Semver version: v${env.VERSION}" // debug
+
+    // git semver gives us the next version due to the pipelines bumping
+    // at the end. We will need to rollback one previous semver
+    // version to get the "current"
+
+    def isPre = nextSemver.contains('-')
+    def currSemver
+
+    if(isPre) {
+        semverSplit  = nextSemver.split('-')
+        def preSplit = semverSplit[1].split('\\.')
+        def nextPre  = preSplit[1].toInteger()
+        def currPre  = nextPre - 1
+
+        currSemver = "${semverSplit[0]}-${preSplit[0]}.${currPre}"
+    } else {
+        def semverSplit = nextSemver.split('\\.')
+        def nextVer = semverSplit.last().toInteger()
+        def currVer = nextVer - 1
+
+        def currSemverArray = []
+        for(int i = 0; i < semverSplit.size(); i++) {
+            def ver = (i != semverSplit.size() - 1) ? semverSplit[i] : currVer
+            currSemverArray << ver
+        }
+        currSemver = currSemverArray.join('.')
+    }
+
+    env.setProperty('VERSION', currSemver)
+    sh "echo '${currSemver}' > VERSION"
+
+    // sh 'cat VERSION' // debug
+    println "--> Current version is: ${env.VERSION}" // debug
 
+    currSemver
 }
```

Comment:
Is this functionality that can live inside edgex-global-pipelines somewhere to be reused across other Jenkinsfiles?

Reply:
Eventually...I think we can bake it into the edgeXSemver function but I did not want to increase scope at the moment.